include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce_then_scan.h

```diff
@@ -1751,7 +1941,21 @@ __parallel_transform_reduce_then_scan(sycl::queue& __q, const std::size_t __n, _
                                           __num_work_groups * __work_group_size);
         }
     }
-    return __future{std::move(__event), std::move(__result_and_scratch)};
+    return __future{std::move(__prior_event), std::move(__result_and_scratch)};
+}
+
+template <typename _CustomName, typename _InRng, typename _GenReduceInput>
+sycl::event
+__parallel_set_balanced_path_partition(sycl::queue& __q, _InRng&& __in_rng, std::size_t __num_diagonals,
+                                       _GenReduceInput __gen_reduce_input)
```

Comment:
```suggestion
__parallel_set_balanced_path_partition(sycl::queue& __q, const _InRng& __in_rng, std::size_t __num_diagonals,
                                       _GenReduceInput __gen_reduce_input)
```

Reply:
As far as I understand, `__in_rng` describes some source data, which is `const` and can't be modified inside this function. So I think better to describe this semantic by `const ref` instead of universal reference.